scripts/tests/test_zkai_d128_recursive_pcd_route_selector_gate.py

```
@@ -0,0 +1,243 @@
+        for mutation in (
+            "route_local_stwo_nested_verifier_relabel_to_go",
+            "route_local_pcd_relabel_to_go",
+            "route_external_adapter_relabel_to_go",
+            "proof_size_metric_smuggled",
+            "verifier_time_metric_smuggled",
+            "proof_generation_time_metric_smuggled",
+            "decision_changed_to_go",
+        ):
+            with self.subTest(mutation=mutation):
+                mutated = GATE.mutate_payload(self.fresh_payload(), mutation)
+                with self.assertRaises(GATE.D128RecursivePCDRouteSelectorError):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)